ironfish/src/migrations/migration.ts

```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import { IronfishNode } from '../node'
+import { IDatabase, IDatabaseTransaction } from '../storage'
+
+export abstract class Migration {
+  id = 0
+  abstract name: string
+
+  init(): Migration {
+    this.id = Number(this.name.split('-')[0])
```

Comment:
Could we just have 2 fields for this so we don't have to do a split?

Reply:
Or have `abstract id: number` as well? Any reason for not doing that?

Reply:
> Or have abstract id: number as well? Any reason for not doing that?

You only declare abstract when you want to force the child to implement it. In this case it's an ID managed by the migration system.

> Could we just have 2 fields for this so we don't have to do a split?

Can you explain what issue you have withi splitting the string? If there's 2 fields, it's 2 fields you need to update in the migration. This gets the name / id from the field.

The alternative could be getting the ID / name from the class but then it won't match the file. Other migration systems. generate it from the file, but we can't exactly do that here.

Reply:
should've added 'nit'. Just thought it would provide better type safety to have 2 fields instead of splitting a string. But I see you're getting it from the filename so that makes sense